sqsworkers/crew.py

```diff
@@ -65,11 +69,14 @@ def __init__(self, **kwargs):
         self.worker_limit = kwargs['worker_limit'] if 'worker_limit' in kwargs else 10
         self.max_number_of_messages = kwargs['max_number_of_messages'] if 'max_number_of_messages' in kwargs else 1
         self.wait_time = kwargs['wait_time'] if 'wait_time' in kwargs else 20
+        # bulk_mode causes sqsworkers to hand over all messages to
+        # MsgProcessor instance at once facilitating bulk operations downstream
+        self.bulk_mode = kwargs['bulk_mode'] if 'bulk_mode' in kwargs else False
         self.exception_handler_function = kwargs['exception_handler'] if 'exception_handler' in kwargs else \
                                           'default_exception_handler'
 
         if not ((self.sqs_session and self.queue_name) or self.sqs_resource):
-            raise TypeError('Required arguments not provided.  Either provide (sqs_session + queue_name) or sqs_resource.')
+            raise TypeError('Required arguments not provided. Either provide (sqs_session + queue_name) or sqs_resource.')
```

Comment:
`ValueError` is the convention for invalid argument functions

`TypeError` is raised when trying to compose disjoint types i.e. `7 + 'pirate'`

```suggestion
            raise ValueError('Required arguments not provided. Either provide (sqs_session + queue_name) or sqs_resource.')
```

Reply:
Including this change.

Reply:
`ValueError` is the convention for invalid argument functions

`TypeError` is raised when trying to compose disjoint types i.e. `7 + 'pirate'`

```suggestion
            raise ValueError('Required arguments not provided. Either provide (sqs_session + queue_name) or sqs_resource.')
```

You said you'd include the change but resolved the conversation without actually doing it